server/src/main/java/org/opensearch/index/IndexSettings.java

```diff
@@ -1211,9 +1212,12 @@ public int getNumberOfReplicas() {
 
     /**
      * Returns true if segment replication is enabled on the index.
+     *
+     * Every shard on a remote node would also have SegRep enabled even without
+     * proper index setting during the migration.
      */
     public boolean isSegRepEnabled() {
-        return ReplicationType.SEGMENT.equals(replicationType);
+        return ReplicationType.SEGMENT.equals(replicationType) || isRemoteNode();
```

Comment:
Did not get the reason for checking remote node to find if an index is seg rep enabled or not?

Reply:
On remote node, index will behave as seg rep enabled with remote with dual replication enabled appropriately.